src/net_processing.cpp

```diff
@@ -4338,9 +4386,24 @@ void PeerManagerImpl::ProcessMessage(CNode& pfrom, const std::string& msg_type,
                     txn.blockhash = blockhash;
                     blockTxnMsg << txn;
                     fProcessBLOCKTXN = true;
-                } else {
+                } else if (first_in_flight) {
+                    // We will try to round-trip any compact blocks we get on failure,
+                    // as long as it's first...
                     req.blockhash = pindex->GetBlockHash();
                     m_connman.PushMessage(&pfrom, msgMaker.Make(NetMsgType::GETBLOCKTXN, req));
+                } else if (pfrom.m_bip152_highbandwidth_to &&
+                    (!nodestate->m_is_inbound ||
```

Comment:
Should be `!pfrom.IsInboundConn()` presumably, to include manual connections at least.

Reply:
took @ajtowns suggestion

Reply:
This doesn't result in a behavior change, does it?  As far as I can tell `nodestate->m_is_inbound` is initialized to the value of `pfrom.IsInboundConn()` when the `CNodeState()` is constructed -- is that understanding correct?

I'm good with avoiding the use of CNodeState where possible though, if that is the reason for this change.

If these are different somehow, then perhaps we'd want to change the logic in `IsBlockRequestedFromOutbound` to match it.

Reply:
> This doesn't result in a behavior change, does it? As far as I can tell nodestate->m_is_inbound is initialized to the value of pfrom.IsInboundConn() when the CNodeState() is constructed -- is that understanding correct?

Yes that is correct, it does not change behavior (my original suggestion would have changed it slightly, as it would exclude manual conns but that wasn't intentional).

> I'm good with avoiding the use of CNodeState where possible though, if that is the reason for this change.

That was the only reason for this nit from my side.